tf5muxserver/mux_server_PrepareProviderConfig.go

```diff
@@ -9,18 +9,20 @@ import (
 
 	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
 	"github.com/hashicorp/terraform-plugin-mux/internal/logging"
-	"github.com/hashicorp/terraform-plugin-mux/internal/tf5dynamicvalue"
 )
 
 // PrepareProviderConfig calls the PrepareProviderConfig method on each server
 // in order, passing `req`. Response diagnostics are appended from all servers.
 // Response PreparedConfig must be equal across all servers with nil values
 // skipped.
-func (s muxServer) PrepareProviderConfig(ctx context.Context, req *tfprotov5.PrepareProviderConfigRequest) (*tfprotov5.PrepareProviderConfigResponse, error) {
+func (s *muxServer) PrepareProviderConfig(ctx context.Context, req *tfprotov5.PrepareProviderConfigRequest) (*tfprotov5.PrepareProviderConfigResponse, error) {
 	rpc := "PrepareProviderConfig"
 	ctx = logging.InitContext(ctx)
 	ctx = logging.RpcContext(ctx, rpc)
-	var resp *tfprotov5.PrepareProviderConfigResponse
+
+	resp := &tfprotov5.PrepareProviderConfigResponse{
+		PreparedConfig: req.Config, // ignored by Terraform anyways
+	}
```

Comment:
This might be the most awkward change to the mux server implementations in this change set -- previously `PrepareProviderConfig`/`ValidateProviderConfig` would check if each underlying provider returned an equal `PreparedConfig` response. This response field was for legacy Terraform usage (e.g. pre-0.12) where a provider could slightly change what core saw as the configuration data for a provider, such as injecting an environment variable value for a "required" provider configuration even though a value was not present in the real configuration. The only higher level provider SDK that enabled that behavior was terraform-plugin-sdk. This response field is ignored by any supported Terraform version that implements protocol versions 5 and 6 short of validating (at least currently) that it is either nil or that data value conforms to the provider schema. Therefore, while it may be safe to wholly omit the value (nil), we can instead use the framework's trick of just roundtripping the request configuration data which is guaranteed to be correct.